### PR TITLE
Rework of Shovel positions

### PR DESCRIPTION
**Pull request by schwiti6190**
-small MP fixes
-fixes Shovel positions in MP
-fixes e tractor for Alpine DLC

### CODE REVIEW

File: TriggerHandler.lua [hunk 1]
```diff
@@ -408,7 +408,9 @@ end
 
 function TriggerHandler:needsFuel()
 	local dieselIndex = self.vehicle:getConsumerFillUnitIndex(FillType.DIESEL)
-	local currentFuelPercentage = self.vehicle:getFillUnitFillLevelPercentage(dieselIndex) * 100
+	local electricIndex = self.vehicle:getConsumerFillUnitIndex(FillType.ELECTRICCHARGE)
```
Comment (pvaiko) [inline TriggerHandler.lua:411]: Can we consolidate this fuel checking into a single function like ADriverUtil.getCurrentFuelPercentage() and use that from here and from AIDriver?

File: TriggerHandler.lua [hunk 2]
```diff
@@ -771,7 +773,7 @@ function TriggerHandler:onActivateObject(superFunc,vehicle)
 			--seperarte for only loading Fuel in to motors!
 			if triggerHandler:isAllowedToLoadFuel() and fillableObject == vehicle then 
 				for fillTypeIndex, fillLevel in pairs(fillLevels) do
-					if fillTypeIndex == FillType.DIESEL  then 
+					if fillTypeIndex == FillType.DIESEL or fillTypeIndex == FillType.ELECTRICCHARGE then
```
Comment (pvaiko) [inline TriggerHandler.lua:776]: and this probably, as isFuel(fillTypeIndex)? (Wondering if Giants did not do that already, have to check if they released the new code)

File: settings.lua [hunk 3]
```diff
@@ -3763,6 +3702,325 @@ function OppositeTurnModeSetting:init(vehicle)
...
+	end
+end
+
+function WorkingToolPositionsSetting:saveToXml(xml, parentKey)
+	setXMLBool(xml, self:getKey(parentKey), self:hasValidToolPositions())
+end
+
+function WorkingToolPositionsSetting:saveToXMLFile(xmlFile, key, usedModNames)
+	local spec = self.spec_cylindered
+	if spec.cpWorkingToolPos == nil or #spec.cpWorkingToolPos < spec.cpWorkingToolPosMax then
```
Comment (pvaiko) [inline settings.lua:3939]: not sure about the history of this, is this for backward compatibility only or are we still storing CP related data in cpWorkingToolPos? What is it storing? And what is the WorkingToolPositionsSetting storing?

**Review by pvaiko — CHANGES_REQUESTED** [review]

File: settings.lua (hunk 3 above)
Comment (schwiti6190) [reply settings.lua:3939]: Every object tool positions gets stored in the object directly as opposed to storing every object position in the main vehicle driver for easy saving and loading of these porsitions. "cpWorkingTool" is just a table I created for each object to store these positions.

File: TriggerHandler.lua (hunk 2 above)
Comment (schwiti6190) [reply TriggerHandler.lua:776]: havn't seen that function yet, but we probably could use this one.

File: TriggerHandler.lua (hunk 1 above)
Comment (schwiti6190) [reply TriggerHandler.lua:411]: Will have a look at this one, could probably consolidate this one with AIDriver:checkFuel().

**Pull request merged**